File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        int n= nums.size();
        unordered_set<int> se;
        for(int i=0;i<n;i++){
            if(se.find(nums[i])!=se.end()){
                return true;
            }
            se.insert(nums[i]);
        }
        return false;

    }
};

File: Data Structures & Algorithms/target-sum/submission-0.py
class Solution:
    def findTargetSumWays(self, nums: List[int], target: int) -> int:
        
        def dfs(i,target):
            if target==0 and i==len(nums):
                return 1
            if i >=len(nums):
                return 0
            
            return dfs(i+1,target-nums[i]) + dfs(i+1,target+nums[i])
        return dfs(0,target)